category/application/command.go

```diff
@@ -0,0 +1,163 @@
+package application
+
+import (
+	"context"
+	"net/url"
+	"strconv"
+
+	"flamingo.me/flamingo/v3/framework/web"
+
+	"flamingo.me/flamingo-commerce/v3/category/domain"
+	"flamingo.me/flamingo-commerce/v3/product/application"
+	searchApplication "flamingo.me/flamingo-commerce/v3/search/application"
+	searchDomain "flamingo.me/flamingo-commerce/v3/search/domain"
+	"flamingo.me/flamingo-commerce/v3/search/utils"
+)
+
+type (
+	// CommandHandler provides the base command logic that is agnostic to the actual view type
+	CommandHandler interface {
```

Comment:
> First of all the Task was so simple - Respond with 404
> The result is a completly refactored controller - application logic - was that really the task of the story? Did that concentrate on the right value generation?

You're absolutely right!

Since the current implementation is not really testable and is also currently not tested, adding a new aspect without a proper way of testing that aspect seemed worse than bloating the task with a refactoring.

> * [...] but thats imho a different task and should involve a general ADR and guidelines

I don't really see a point for an ADR just for a basic concept such as the command pattern. But true, this might be better in a separate change. We could easily revert the last refactoring commit and do it separately.

> * [...] adding interfaces and default implementation for "everything"

Well, abstraction is the only way for clean and easy testability.

> * Now there are  many aspects in application layer that belong to the "web" interface [...]

I was aware of this and wasn't very happy about it either. This is not clean and would need further refactoring.

> The question is how much stays? Potential reusability should be given with the graphql interface - then it even makes sense to refactor now

I was thinking about this as well and thought this could be done in subsequent refactoring.

But maybe, the last refactoring ("command pattern") was too much. How about I revert this and we look into doing this afterwards?